databases/week1/db-week1-hw.sql

```diff
@@ -0,0 +1,58 @@
+1. Find out how many tasks are in the task table
+SELECT COUNT(*) FROM task;
+
+2. Find out how many tasks in the task table do not have a valid due date
+SELECT COUNT(*) FROM task WHERE due_date IS NULL;
+
+
+3. Find all the tasks that are marked as done
+SELECT task.id, task.title, status.name, status.id FROM task, status where status.name LIKE "Done" 
+AND task.status_id=status.id;
+
+using Join
+SELECT t.id,t.title,t.description,t.status_id,t.user_id 
+FROM task t JOIN status ON t.status_id = status.id
+WHERE status.name = "Done";
+
+
+4. Find all the tasks that are not marked as done
+SELECT task.id, task.title, status.name, status.id FROM task, status WHERE status.name NOT LIKE "Done" AND task.status_id=status.id;
+
+using Join
+SELECT t.id,t.title,t.description,t.status_id,t.user_id 
+FROM task t JOIN status ON t.status_id = status.id
+WHERE status.name NOT LIKE "Done";
+
+5. Get all the tasks, sorted with the most recently created first
+SELECT id, title, description, created FROM task ORDER BY created DESC;
+
+6. Get the single most recently created task
+SELECT id, title, description, created FROM task ORDER BY created DESC LIMIT 1;
+
+
+7. Get the title and due date of all tasks where the title or description contains database
+SELECT title, due_date FROM task WHERE title LIKE "%database%" OR description LIKE "%database%";
+
+8. Get the title and status (as text) of all tasks
+SELECT task.title, status.name FROM task, status WHERE task.status_id=status.id;
+
+using Join
+SELECT task.title,status.name AS text
+FROM task INNER JOIN status ON task.status_id = status.id;
```

Comment:
Why INNER JOIN here and only JOIN clause in other lines. Read the difference between those Why you are using the JOIN clause here? No difference between JOIN & INNER JOIN. So try to use one of them for good practice.
https://blog.sqlauthority.com/2013/07/16/sql-difference-between-inner-join-and-join/